From 946e0dde4be61eb114e6080fce9f0f8e34152e34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Thu, 4 Apr 2013 14:40:47 +0200
Subject: [PATCH 033/127] wined3d: Shadow device->offscreenBuffer in the
 context
Reply-To: wine-devel <wine-devel@winehq.org>

Meh. Should probably go into the adapter, or non-fbo ORM should just die
TODO 2: See what this is actually used for.
---
 dlls/wined3d/context.c         | 2 ++
 dlls/wined3d/device.c          | 1 +
 dlls/wined3d/wined3d_private.h | 2 ++
 3 files changed, 5 insertions(+)

diff --git a/dlls/wined3d/context.c b/dlls/wined3d/context.c
index 861b812..dc867b2 100644
--- a/dlls/wined3d/context.c
+++ b/dlls/wined3d/context.c
@@ -1358,6 +1358,8 @@ struct wined3d_context *context_create(struct wined3d_swapchain *swapchain,
     ret->current_fb.rt_size = gl_info->limits.buffers;
     if (!ret->current_fb.render_targets)
         goto out;
+    if (device->context_count)
+        ret->offscreenBuffer = device->contexts[0]->offscreenBuffer;
 
     /* Initialize the texture unit mapping to a 1:1 mapping */
     for (s = 0; s < MAX_COMBINED_SAMPLERS; ++s)
diff --git a/dlls/wined3d/device.c b/dlls/wined3d/device.c
index 3bcf1ab..0d7ca02 100644
--- a/dlls/wined3d/device.c
+++ b/dlls/wined3d/device.c
@@ -922,6 +922,7 @@ HRESULT CDECL wined3d_device_init_3d(struct wined3d_device *device,
             }
         }
     }
+    device->contexts[0]->offscreenBuffer = device->offscreenBuffer;
 
     TRACE("All defaults now set up, leaving 3D init.\n");
 
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index ac5e372..61d270e 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -1184,6 +1184,8 @@ struct wined3d_context
     GLfloat                 fog_coord_value;
     GLfloat                 color[4], fogstart, fogend, fogcolor[4];
     GLuint                  dummy_arbfp_prog;
+
+    GLenum                  offscreenBuffer;
 };
 
 typedef void (*APPLYSTATEFUNC)(struct wined3d_context *ctx, const struct wined3d_state *state, DWORD state_id);
-- 
1.8.1.5

